src/stdatamodels/model_base.py

```diff
@@ -646,14 +646,10 @@ def shape(self):
                 self._shape = primary_array.shape
         return self._shape
 
-    def my_attribute(self, attr):
-        properties = frozenset(("shape", "history", "_extra_fits", "schema"))
-        return attr in properties
-
     def __setattr__(self, attr, value):
-        if self.my_attribute(attr):
+        if attr in frozenset(("shape", "history", "_extra_fits", "schema")):
             object.__setattr__(self, attr, value)
-        elif ndmodel.NDModel.my_attribute(self, attr):
+        elif attr in frozenset(("data", "mask", "unit", "wcs", "unceratainty")):
```

Comment:
While NDModel still exists it seems like it should own this list.  Maybe we just rename the method to `_my_attribute` for now?